src/main/java/bio/terra/common/FutureUtils.java

```diff
@@ -31,56 +31,52 @@ public static <T> List<T> waitFor(final List<Future<T>> futures) {
    * @return The resolved values of the futures. There is no guarantee that the order of returned
    *     values matches the order of the passed in futures
    */
-  public static <T> List<T> waitFor(
-      final List<Future<T>> futures, final Optional<Duration> maxThreadWait) {
+  public static <T> List<T> waitFor(final List<Future<T>> futures, final Duration maxThreadWait) {
 
-    final AtomicReference<Optional<ApiException>> foundFailure =
-        new AtomicReference<>(Optional.empty());
+    final AtomicReference<ErrorReportException> foundFailure = new AtomicReference<>();
```

Comment:
Would it be worthwhile to use a concurrent list here to collect all errors across all threads?

Reply:
I don't think so since by definition if doesn't wait for the other errors to throw. It just waits for one and then cancels the rest